### PR TITLE
RavenDB-7886 load only one settings.json file

### PR DESCRIPTION
**Pull request by gregolsky**
- move platform dependant release mode settings to Properties dir
- copy proper one in build.ps1
- when using custom config dont add settings.json from server dir
- adjust build scripts for the above

TODO after merge adjust teamcity settings for the new path

### CODE REVIEW

File: src/Raven.Server/Config/RavenConfiguration.cs [hunk 1]
```diff
@@ -103,18 +103,15 @@ private void AddJsonConfigurationVariables(string customConfigPath = null)
         {
             if (string.IsNullOrEmpty(customConfigPath) == false)
             {
-                    _configBuilder.AddJsonFile(customConfigPath, optional: true);
-                // if we were specified a non existing file, don't use the defaults
-                return;
+                if (File.Exists(customConfigPath) == false)
+                    throw new FileNotFoundException("Custom configuration file has not been found.", customConfigPath);
```
Comment (ayende) [inline src/Raven.Server/Config/RavenConfiguration.cs:107]: We are explicitly passing non existent files sometimes in tests.

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Config/RavenConfiguration.cs (hunk 1 above)
Comment (gregolsky) [reply src/Raven.Server/Config/RavenConfiguration.cs:107]: OK got it, let me check.

Comment (ayende) [reply src/Raven.Server/Config/RavenConfiguration.cs:107]: `Default_database_path_settings` and friends

Comment (gregolsky) [reply src/Raven.Server/Config/RavenConfiguration.cs:107]: Fixed the tests to use an empty settings file. We'd like the user to know if he messed up the config file path.

**Pull request merged**